modules/apps/27-interchain-accounts/host/keeper/handshake_test.go

```diff
@@ -95,6 +95,27 @@ func (suite *KeeperTestSuite) TestOnChanOpenTry() {
 				suite.Require().False(found)
 			}, true,
 		},
+		{
+			"success - previous metadata is different",
+			func() {
+				// create a new channel and set it in state
+				ch := channeltypes.NewChannel(channeltypes.CLOSED, channeltypes.ORDERED, channeltypes.NewCounterparty(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID), []string{path.EndpointA.ConnectionID}, TestVersion)
+				suite.chainB.GetSimApp().GetIBCKeeper().ChannelKeeper.SetChannel(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, ch)
+
+				// set the active channelID in state
+				suite.chainB.GetSimApp().ICAHostKeeper.SetActiveChannelID(suite.chainB.GetContext(), path.EndpointB.ConnectionID, path.EndpointA.ChannelConfig.PortID, path.EndpointB.ChannelID)
+
+				// set the previous encoding to be proto3json. And then set the new encoding to be protobuf.
+				metadata.Encoding = icatypes.EncodingProto3JSON
```

Comment:
> And then set the new encoding to be protobuf.

This is because `path.EndpointA.ChannelConfig.Version` is passed as `counterpartyVersion` on L312, correct? 
Maybe we could make that more explicit in the comment?